### DISCUSSION TITLE
请问tcp新块通知怎么设置，谢谢。

### DISCUSSION THREAD

**Discussion by ruyichang** [nervosnetwork/ckb#2431] [General]
如题

**Comment by ruyichang** [comment]
或者有没有其他形式的新高度通知，类似btc的zmq等。

**Comment by quake** [comment]
首先需要在 ckb.toml 配置
```
tcp_listen_address = "127.0.0.1:18114"
```

然后通过 rpc 订阅，具体文档：
https://github.com/nervosnetwork/ckb/tree/develop/rpc#subscription

**Comment by quake** [comment]
> 或者有没有其他形式的新高度通知，类似btc的zmq等。

还有一种方式是 ckb .toml 配置 notifier 里面的 new_block_notify_script: 
```
# [notifier]
# # Execute command when the new tip block changes, first arg is block hash.
# new_block_notify_script = "your_new_block_notify_script.sh"
```

**Comment by ruyichang** [comment]
> 首先需要在 ckb.toml 配置
> 
> ```
> tcp_listen_address = "127.0.0.1:18114"
> ```
> 
> 然后通过 rpc 订阅，具体文档：
> https://github.com/nervosnetwork/ckb/tree/develop/rpc#subscription

最理想的情况是可以通过subscribe订阅最新的get_block_template，请问目前有这个打算吗？

**Comment by quake** [comment]
你是在做挖矿的相关功能？你可以订阅 new_tip_header，一旦有新的 header 通知，就再去调用一下 get_block_template 

或者通过 new_block_notify_script 这个脚本去回调你自己的服务，让它去刷新 get_block_template

这样都能让 miner 第一时间工作在最新的 block template 上
